cmd/account/account-list.go

```diff
@@ -0,0 +1,179 @@
+	accountListCmd.Flags().StringVarP(&ops.username, "user", "u", "", "LDAP username")
+	accountListCmd.Flags().StringVarP(&ops.payerAccount, "payer-account", "p", "", "Payer account type")
+	accountListCmd.Flags().StringVarP(&ops.accountID, "account-id", "i", "", "Account ID")
+
+	return accountListCmd
+}
+
+func (o *accountListOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.payerAccount == "" {
+		return cmdutil.UsageErrorf(cmd, "Payer account was not provided")
+	}
+	if o.username != "" && o.accountID != "" {
+		return cmdutil.UsageErrorf(cmd, "Cannot provide both username and account ID")
+	}
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *accountListOptions) run() error {
+
+	var (
+		tempAccountIDs []string
+		user           string
+		OuID           string
+	)
+	// Instantiate Aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	if o.payerAccount == "osd-staging-2" {
+		OuID = "ou-rs3h-ry0hn2l9"
+	} else if o.payerAccount == "osd-staging-1" {
+		OuID = "ou-0wd6-z6tzkjek"
+	}
+
+	input := &organizations.ListAccountsForParentInput{
+		ParentId: &OuID,
+	}
+	accounts, err := awsClient.ListAccountsForParent(input)
+	if err != nil {
+		return err
+	}
+	// Initialize hashmap
+	m := make(map[string][]string)
+
+	if o.username == "" {
+		// Loop through list of accounts and build hashmap of users and accounts
+		for _, a := range accounts.Accounts {
+			inputListTags := &organizations.ListTagsForResourceInput{
+				ResourceId: a.Id,
+			}
+			tagVal, err := awsClient.ListTagsForResource(inputListTags)
+			if err != nil {
+				return err
+			}
+			for _, t := range tagVal.Tags {
+				if t.Key == aws.String("owner") {
+					user = *t.Value
+				}
+			}
+			// Check if user is already in the hashmap
+			// If yes append to list of accounts
+			// If no create list of accounts and add it as value to key
+			_, ok := m[user]
+			if ok {
+				_ = append(m[user], *a.Id)
+			} else {
+				tempAccountIDs = append(tempAccountIDs, *a.Id)
+				m[user] = tempAccountIDs
+			}
+		}
+	} else if o.username != "" {
+		user = o.username
+		// Create input to list the accounts from a specific user
+		inputFilterTag := &resourcegroupstaggingapi.GetResourcesInput{
+			TagFilters: []*resourcegroupstaggingapi.TagFilter{
+				{
+					Key: aws.String("owner"),
+					Values: []*string{
+						aws.String(user),
+					},
+				},
+			},
+		}
+		accounts, err := awsClient.GetResources(inputFilterTag)
+		if err != nil {
+			return err
+		}
+
+		// Get last 9 digits of ResourceARN and append it to account list
+		for _, a := range accounts.ResourceTagMappingList {
+			tempAccountIDs = append(tempAccountIDs, (*a.ResourceARN)[len(*a.ResourceARN)-9:])
```

Comment:
AWS accounts are 12 characters long, this needs to be changed.

We could probably split on `/` and get the last value for some "future-proofing" in case AWS changes that in the future, but I'll leave that decision up to you.